src/commands/process_metavaults.py

```diff
@@ -0,0 +1,202 @@
+    help='JWT secret key used for signing and verifying JSON Web Tokens'
+    ' when connecting to execution nodes.',
+)
+@click.option(
+    '--consensus-endpoints',
+    type=str,
+    envvar='CONSENSUS_ENDPOINTS',
+    prompt='Enter the comma separated list of API endpoints for consensus nodes',
+    help='Comma separated list of API endpoints for consensus nodes.',
+)
+@click.option(
+    '--graph-endpoint',
+    type=str,
+    envvar='GRAPH_ENDPOINT',
+    help='API endpoint for graph node.',
+)
+@click.option(
+    '--log-format',
+    type=click.Choice(
+        LOG_FORMATS,
+        case_sensitive=False,
+    ),
+    default=LOG_PLAIN,
+    envvar='LOG_FORMAT',
+    help='The log record format. Can be "plain" or "json".',
+)
+@click.option(
+    '--log-level',
+    type=click.Choice(
+        LOG_LEVELS,
+        case_sensitive=False,
+    ),
+    default='INFO',
+    envvar='LOG_LEVEL',
+    help='The log level.',
+)
+@click.option(
+    '--network',
+    help='The network of the vault. Default is the network specified at "init" command.',
+    type=click.Choice(
+        AVAILABLE_NETWORKS,
+        case_sensitive=False,
+    ),
+)
+@click.command(
+    help='Processes one or more meta vaults by '
+    'performing state updates and handling deposit and exit requests.'
+)
+# pylint: disable-next=too-many-arguments,too-many-locals
+def process_metavaults(
+    vaults: str,
+    consensus_endpoints: str,
+    execution_endpoints: str,
+    execution_jwt_secret: str | None,
+    graph_endpoint: str,
+    network: str,
+    verbose: bool,
+    log_level: str,
+    log_format: str,
+    wallet_file: str | None,
+    wallet_password_file: str | None,
+    max_fee_per_gas_gwei: int | None,
+    meta_vault_min_deposit_amount_gwei: int,
+    meta_vault_update_interval: int,
+) -> None:
+    vault_addresses = [to_checksum_address(address) for address in vaults.split(',')]
+    settings.set(
+        # mock vault and vault_dir
+        vault=vault_addresses[0],
+        vault_dir=Path.home() / '.stakewise',
+        consensus_endpoints=consensus_endpoints,
+        execution_endpoints=execution_endpoints,
+        execution_jwt_secret=execution_jwt_secret,
+        graph_endpoint=graph_endpoint,
+        verbose=verbose,
+        network=network,
+        wallet_file=wallet_file,
+        wallet_password_file=wallet_password_file,
+        max_fee_per_gas_gwei=max_fee_per_gas_gwei,
+        log_level=log_level,
+        log_format=log_format,
+        meta_vault_min_deposit_amount_gwei=Gwei(meta_vault_min_deposit_amount_gwei),
+        meta_vault_update_interval=meta_vault_update_interval,
+    )
+    try:
+        asyncio.run(main(vault_addresses))
+    except Exception as e:
+        log_verbose(e)
+        sys.exit(1)
+
+
+async def main(vaults: list[ChecksumAddress]) -> None:
+    setup_logging()
+    setup_sentry()
+    await setup_clients()
+
+    log_start()
+
+    chain_state = await get_chain_finalized_head()
```

Comment:
I suggest to use head instead as we don't have state here, so no need to worry about forks.